File: JavaArchitecture/app/src/main/java/com/demo/architecture/injection/module/ActivityModule.java
package com.demo.architecture.injection.module;

import android.app.Activity;
import android.content.Context;


import com.demo.architecture.injection.ActivityContext;

import dagger.Module;
import dagger.Provides;

/**
 * Created by HeYingXin on 2018/4/17.
 */
@Module
public class ActivityModule{

    private Activity mActivity;

    public ActivityModule(Activity activity){
        mActivity = activity;
    }

    @Provides
    Activity provideActivity(){
        return mActivity;
    }

    @Provides
    @ActivityContext
    Context provideContext(){
        return mActivity;
    }
}

File: kotlin-frist-branch-TODO/app/src/main/java/com/mazouri/mvpkotlin/injection/module/ActivityModule.kt
package com.mazouri.mvpkotlin.injection.module

import android.app.Activity
import android.content.Context
import com.mazouri.mvpkotlin.data.GithubService
import com.mazouri.mvpkotlin.data.GithubServiceFactory
import com.mazouri.mvpkotlin.injection.ActivityContext
import com.mazouri.mvpkotlin.injection.ApplicationContext
import com.mazouri.mvpkotlin.task.main.MainPresenter
import dagger.Module
import dagger.Provides

/**
 * Created by HeYingXin on 2018/4/17.
 */
@Module
class ActivityModule(private val mActivity: Activity) {

    @Provides
    internal fun provideActivity(): Activity {
        return mActivity
    }

    @Provides
    @ActivityContext
    internal fun provideContext(): Context {
        return mActivity
    }

    @Provides
    @ActivityContext
    internal fun provideGithubService(): GithubService {
        return GithubServiceFactory.makeStarterService()
    }
}